app/models/create_expenses_with_receipt.rb

```diff
@@ -0,0 +1,31 @@
+class CreateExpensesWithReceipt
+  attr_reader :expense
+
+  def initialize(expense_params)
+    @params = expense_params
+  end
+
+  def create
+    return false unless picture_exists?
+
+    Expense.transaction do
+      create_expense
+      create_receipt
+    end
+    @expense.persisted?
+  end
```

Comment:
What raises that error? You'll probably have to catch that error and return false, with the proper errors set up.